Adding benchmarking info to collections.md (#2236)
near/docs: docs/2.build/2.smart-contracts/anatomy/collections.md

Understanding how the contract stores and loads both types of collections is crucial to decide which one to use.

<details>

<summary> Native vs SDK Collections </summary> 

Use native collections for small amounts of data that need to be accessed all together, and SDK collections for large amounts of data that do not need to be accessed all together.

> If your collection has up to 100 entries, it's acceptable to use the native collection, as it might be simpler since you don't have to manage prefixes as we do with SDK collections.

Reviewer: ```suggestion

If your collection has up to 100 entries, it's acceptable to use the native collection, as it might be simpler since you don't have to manage prefixes as we do with SDK collections.
```

> However, if your collection has 1,000 or more entries, it's better to use SDK collection. The [investigation](https://github.com/volodymyr-matselyukh/near-benchmarking) shows that running the `contains` method on a native HashSet<i32> consumes 41% more gas compared to SDK IterableSet<i32>.

Reviewer: ```suggestion

However, if your collection has 1,000 or more entries, it's better to use SDK collection. [This user test](https://github.com/volodymyr-matselyukh/near-benchmarking) shows that running the `contains` method on a native HashSet<i32> consumes 41% more gas compared to SDK IterableSet<i32>.
```

Understanding how the contract stores and loads both types of collections is crucial to decide which one to use.

<details>

<summary> Native vs SDK Collections </summary> 

Use native collections for small amounts of data that need to be accessed all together, and SDK collections for large amounts of data that do not need to be accessed all together.

If your collection has up to 100 entries, it's acceptable to use the native collection, as it might be simpler since you don't have to manage prefixes as we do with SDK collections.

However, if your collection has 1,000 or more entries, it's better to use SDK collection. [This user test](https://github.com/volodymyr-matselyukh/near-benchmarking) shows that running the `contains` method on a native HashSet<i32> consumes 41% more gas compared to SDK IterableSet<i32>.

</details>

> :::info How the State is Handled

Reviewer: ```suggestion
<details>

 <summary> How the State is Handled </summary>
```

> Each time the contract is executed, the first thing it will do is to read the values and [deserialize](./serialization.md) them into memory, and after the function finishes, it will [serialize](./serialization.md) and write the values back to the database. That means the contract will load your native collections fully into memory before the contract's method execution. The method you invoke may not even use the loaded collection. This will have impact on GAS you spend for methods in your contract. So, using native collection which will have more than 100 entries as the top level property of your contract is a bad practice.

Reviewer: ```suggestion
Each time the contract is executed, the first thing it will do is to read the values and [deserialize](./serialization.md) them into memory, and after the function finishes, it will [serialize](./serialization.md) and write the values back to the database.

That means the contract will load your native collections fully into memory before the contract's method execution. The method you invoke may not even use the loaded collection. This will have impact on GAS you spend for methods in your contract. So, using native collection which will have more than 100 entries as the top level property of your contract is a bad practice.
```

> :::

Reviewer: ```suggestion
</details>
```

> :::tip

Reviewer: ```suggestion
<details>

<summary> Native vs SDK Collections </summary> 
```

Use native collections for small amounts of data that need to be accessed all together, and SDK collections for large amounts of data that do not need to be accessed all together.
If your collection has up to 100 entries, it's acceptable to use the native collection, as it might be simpler since you don't have to manage prefixes as we do with SDK collections.
However, if your collection has 1,000 or more entries, it's better to use SDK collection. The [investigation](https://github.com/volodymyr-matselyukh/near-benchmarking) shows that running the `contains` method on a native HashSet<i32> consumes 41% more gas compared to SDK IterableSet<i32>.

> :::

Reviewer: ```suggestion
</details>
```
